app/android/runtime_client/src/org/xwalk/app/runtime/extension/JsStubGenerator.java

```diff
@@ -47,29 +52,104 @@
      */
     String generate() {
         String result = "";
+        MemberInfo entry = reflection.getEntryPoint();
+        if (entry != null) {
+            result = generateEntryPoint(entry);
+        }
+
+        result = (result.length() > 0) ? result : jsHeader;
         if (reflection.getEventList() != null) {
-            result += generateEventTarget();
+            result += generateEventTarget(reflection);
         }
 
         Map<String, MemberInfo> members = reflection.getMembers();
         for (String key : members.keySet()) {
             MemberInfo m = members.get(key);
+            if (m.isEntryPoint) continue;
             switch (m.type) {
                 case JS_PROPERTY:
-                    result += generateProperty(key, m);
+                    result += generateProperty(m);
                     break;
                 case JS_METHOD:
-                    result += generateMethod(key, m);
+                    result += generateMethod(m, true);
+                    break;
+                case JS_CONSTRUCTOR:
+                    result += generateConstructor(m, true);
                     break;
                 default:
                     break;
             }
         }
-        return jsHeader + result + "\n";
+        return result + "\n";
+    }
+    String generateEntryPoint(MemberInfo entry) {
+        // Is a binding Object.
+        if (entry.type == MemberType.JS_PROPERTY) {
+            Class<?> type = ((Field)(entry.accesser)).getType();
+            String funcName = type.getSimpleName();
+            return jsHeader + String.format("%s(exports, helper);\n", getPrototypeName(funcName));
+
+        } else if (entry.type == MemberType.JS_METHOD) {
+            return String.format("exports = %s;\n %s\n %s",
+                    getInternalName(entry.jsName), jsHeader, generateMethod(entry, false));
+
+        } else if (entry.type == MemberType.JS_CONSTRUCTOR) {
+            return String.format("exports = %s;\n %s\n %s",
+                   entry.jsName, jsHeader, generateConstructor(entry, false));
+        }
+        return "";
```

Comment:
suggest rewrite above logic as to avoid if-else.
```
if (cond1) {
  ...
  return ...;
}

if (cond2) {
  ...
  return ...;
}

return "";
```